cinema/tests/test_movie_view_set.py

```diff
@@ -0,0 +1,130 @@
+        self.admin_token = Token.objects.create(user=self.admin_user)
+        self.client.credentials(HTTP_AUTHORIZATION=f'Token {self.token.key}')
+
+        # Створення жанрів
+        self.genre1 = create_genre("Action")
+        self.genre2 = create_genre("Drama")
+
+        # Створення акторів
+        self.actor1 = create_actor("John", "Doe")
+        self.actor2 = create_actor("Jane", "Smith")
+
+        # Створення фільмів
+        self.movie1 = create_movie(
+            "Movie One", genres=[self.genre1], actors=[self.actor1]
+        )
+        self.movie2 = create_movie(
+            "Movie Two", genres=[self.genre2], actors=[self.actor2]
+        )
+
+    def test_list_movies(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 2)
+
+    def test_filter_movies_by_title(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"title": "Movie One"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_filter_movies_by_genres(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"genres": f"{self.genre1.id}"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_filter_movies_by_actors(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url, {"actors": f"{self.actor1.id}"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], "Movie One")
+
+    def test_create_movie_as_authenticated_user(self):
+        url = reverse("cinema:movie-list")
+        data = {
+            "title": "New Movie",
+            "description": "A new movie description",
+            "duration": 90,
+            "genres": [self.genre1.id],
+            "actors": [self.actor1.id, self.actor2.id],
+        }
+        response = self.client.post(url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_create_movie_as_admin(self):
+        self.client.credentials(HTTP_AUTHORIZATION=f'Token {self.admin_token.key}')
+        url = reverse("cinema:movie-list")
+        data = {
+            "title": "New Movie",
+            "description": "A new movie description",
+            "duration": 90,
+            "genres": [self.genre1.id],
+            "actors": [self.actor1.id, self.actor2.id],
+        }
+        response = self.client.post(url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```

Comment:
This is a bug. In the test_create_movie_as_admin test method, you are not checking if the movie was actually created. After the request, you should query the database to make sure that the movie count has increased by one.